lib/entities.js

```diff
@@ -0,0 +1,231 @@
+    // value uniquely identifies an entity, so its shard key is `internalId`
+    collection: 'tokenization-entity',
+    fields: {'entity.internalId': 1},
+    options: {unique: true, background: false}
+  }, {
+    // automatically expire entities with an `expires` date field
+    collection: 'tokenization-entity',
+    fields: {'entity.expires': 1},
+    options: {
+      unique: false,
+      background: false,
+      expireAfterSeconds: 0
+    }
+  }]);
+});
+
+/**
+ * Gets an entity record identified by `internalId`.
+ *
+ * @param {object} options - Options to use.
+ * @param {Buffer} options.internalId - The internal ID for the entity.
+ *
+ * @returns {Promise<object>} The entity record.
+ */
+export async function get({internalId} = {}) {
+  const collection = database.collections['tokenization-entity'];
+  const query = {'entity.internalId': internalId};
+  const projection = {_id: 0};
+  const record = await collection.findOne(query, {projection});
+  if(!record) {
+    throw new BedrockError(
+      'Entity not found.',
+      'NotFoundError', {
+        httpStatusCode: 404,
+        public: true
+      });
+  }
+  return record;
+}
+
+/**
+ * Sets the `minAssuranceForResolution` for the entity identified by
+ * `internalId`. This value affects whether or not a token that was issued
+ * from an unpinned token batch will resolve; it does not affect the resolution
+ * of tokens that were issued from pinned token batches.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.internalId - The internal ID for the entity.
+ * @param {number} options.minAssuranceForResolution - Minimum level of
+ *   identity assurance required for resolution of tokens that were issued
+ *   from an unpinned token batch.
+ *
+ * @returns {Promise<boolean>} Returns true if update occurred.
+ */
+export async function setMinAssuranceForResolution({
+  internalId, minAssuranceForResolution
+} = {}) {
+  const query = {'entity.internalId': internalId};
+  const $set = {
+    'meta.updated': Date.now(),
+    'entity.minAssuranceForResolution': minAssuranceForResolution
+  };
+
+  const collection = database.collections['tokenization-entity'];
+  const result = await collection.updateOne(
+    query, {$set}, database.writeOptions);
+
+  // return `true` if the update occurred
+  return result.result.n !== 0;
+}
+
+/**
+ * Increments an entity's `batchInvalidationCount` to indicate that all
+ * unpinned batches are about to be invalidated -- and any that are
+ * concurrently created should be considered invalid. The update will only
+ * be applied if the entity exists and its `batchInvalidationCount` has not
+ * changed from the given value. This is to ensure that any decision made to
+ * invalidate token batches based on a previously read value will be consistent
+ * with the current state.
+ *
+ * @param {object} options - Options to use.
+ * @param {string} options.entity - The entity.
```

Comment:
The entity should be an object here.

Reply:
Thanks! Fixed in c4336da90e23b8c91bd395a755a3bfa42299b1d3.